src/bag3_digital/layout/stdcells/and_complex.py

```diff
@@ -381,6 +383,522 @@ def draw_layout(self) -> None:
         )
 
 
+# TODO: port to higher level
```

Comment:
These classes have many BAG2 style sections that need to be updated. E.g. missing return types on various methods, not using f-strings, etc. Please fix all those first. I would also put this in an independent python file.